models/demos/falcon7b/tests/unit_tests/test_falcon_lm_head_matmul_2d.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from loguru import logger
+import pytest
+import torch
+
+import tt_lib as ttl
+
+from models.demos.falcon7b.tt.model_utils import falcon_lm_head_matmul_2d
+from models.utility_functions import comp_pcc, tt2torch_tensor
+
+
+def run_falcon_lm_head_matmul_2d(
+    seq_len, device, in0_mem_config, in1_mem_config, out_mem_config, in0_dtype, in1_dtype, out_dtype
+):
+    pcc = 0.99
+    torch.manual_seed(1234)
+
+    M = seq_len // 32
+    K = 142
+    N = 2032
+
+    a_shape = [1, 1, M * 32, K * 32]
+    # weights are padded on model initialization to reach 144 inner dim
+    b_shape = [1, 1, (K + 2) * 32, N * 32]
+    expected_output_shape = [1, 1, M * 32, N * 32]
+
+    # calculate parameters for the given sequence length
+    num_slices = 4 if seq_len <= 1024 else 8
+
+    # --- prepare activations
+    A = torch.randn(a_shape)
+    padding = torch.zeros([1, 1, M * 32, 64])
+    A_padded = torch.cat([A, padding], -1)
+
+    # padd in0 tensor to have last dimension 144 * 32 instead of 142 * 32
+    a_t = ttl.tensor.Tensor(A, in0_dtype).to(ttl.tensor.Layout.TILE).to(device, in0_mem_config)
```

Comment:
I've seen tt_lib usually using torch2tt_tensor util function to get tensor from torch to device